distance-method/distance_method.py

```diff
@@ -153,28 +156,117 @@ def preprocess_data(working_dir,
             if verbose:
                 sys.stdout.write("%s. %s\t" % (
                     species+1, basename(_file)))
-            with open(_file, 'rb') as readfile:
-                for gene, (label, seq) in enumerate(parse_fasta(readfile)):
-                    label = label.split()[0]
-                    ref_db[label] = seq
-                    sudo_label = "%s_%s" % (species, gene)
-                    if label in gene_map:
-                        raise ValueError("Duplicate sequence labels are "
-                                         "not allowed: %s" % label)
-                    gene_map[label] = sudo_label
-                    gene_map[sudo_label] = label
-                if verbose:
-                    sys.stdout.write("%s\n" % gene)
+            num_genes = 0
+            for gene, seq in enumerate(skbio.io.read(_file, format='fasta')):
```

Comment:
My concern with this is that you will be loading the full gene_map (fasta) into memory, not sure how safe this is. Note that in the past I have seen memory issues do to this ...